cpp/src/strings/replace/multi_re.cu

```diff
@@ -71,35 +75,51 @@ struct replace_multi_regex_fn
         auto nchars = d_str.length();     // number of characters in input string
```

Comment:
I feel like this function  could be parallelized on characters of  all strings rather than  parallelized on  strings.  But that is future work...